blueprint/core/blueprint_View.cpp

```diff
@@ -13,49 +13,67 @@ namespace blueprint
 
     namespace detail
     {
+        juce::var getMouseEventRelatedTarget(const juce::MouseEvent& e, const blueprint::View& view)
+        {
+            juce::Component *topParent              = view.getTopLevelComponent();
+            const juce::MouseEvent topRelativeEvent = e.getEventRelativeTo(topParent);
+
+            juce::Component *componentUnderMouse = topParent->getComponentAt(topRelativeEvent.x, topRelativeEvent.y);
+
+            if (auto v = dynamic_cast<blueprint::View*>(componentUnderMouse))
+                return v->getViewId();
+
+            // return null relatedTarget if event occurred from a non-View component.
+            return {};
+        }
 
         /** A little helper for DynamicObject construction. */
-        juce::var makeViewEventObject (const juce::NamedValueSet& props)
+        juce::var makeViewEventObject (const juce::NamedValueSet& props, const blueprint::View& view)
         {
             auto* o = new juce::DynamicObject();
 
             for (auto& pair : props)
                 o->setProperty(pair.name, pair.value);
 
+            o->setProperty("target", view.getViewId());
+
+            //TODO: Add timeStamp.
```

Comment:
We could also trivially add the timestamp in js if that's easier

Reply:
Yeah totally. Should be a separate PR but whatever works best. Ideally the most performant of the two perhaps so that we can keep the overhead of event creation as low as possible.